/*
 * netsiege - A multiplayer point and click adventure
 * Copyright (C) 2014 Gregor Vollmer <gregor@celement.de>
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 */

package script.pb;

message Property {
    required string name = 1;
    required string decl = 2;
    optional uint32 primitive_size = 3;
    repeated uint64 primitve_dat = 4;
    repeated string string_dat = 5;
    optional ScriptObject object_dat = 6;
}

message ScriptObject {
    repeated Property properties = 1;
    required string object_type = 2;
}
